src/router/storeSync/SearchParamConfig.class.js

```diff
@@ -18,13 +19,37 @@ function dispatchSearchFromUrl(to, store, urlParamValue) {
     })
 }
 
+/**
+ * This will remove the query param from the URL It is necessary to do this in vanilla JS because
+ * the router does not provide a way to remove a query without reloading the page which then removes
+ * the value from the store.
+ *
+ * @param {Object} key The key to remove from the URL
+ */
+function removeQueryParamFromHref(key) {
```

Comment:
What if we start the app with a hash? such as `http://localhost:8080/#/map?swissearch=test`
Will it work too?

Reply:
yes it still works